api/src/main/java/jakarta/data/repository/Select.java

```diff
@@ -25,23 +25,27 @@
 import java.lang.annotation.Target;
 
 /**
- * <p>Determines which entity attribute(s) are returned by a {@link Find}
- * operation.</p>
+ * <p>Determines the attribute or attributes of an entity which are returned
+ * by a {@linkplain Find parameter-based} or {@linkplain Query annotated}
+ * query method.</p>
  *
- * <p>This annotation can be specified on a repository find method or
- * on a record component.</p>
+ * <p>This annotation may be applied to a repository method annotated
+ * {@code @Find} or to a component of a record type.</p>
  *
- * <p>When used on a repository find method that returns an entity attribute
- * type, at most one {@code Select} annotation can annotate the method.</p>
- *
- * <p>When used on a repository find method that returns a Java record type,
- * multiple {@code Select} annotations may annotate the method, one for each
- * record component. The number and order of {@code Select} annotations must
- * match the number and order of record components.</p>
- *
- * <p>When used on a record component, the {@code Select} annotation
- * value is used for all repository methods that lack this
- * annotation and for which the record is the result type.</p>
+ * <ul>
+ * <li>At most one {@code @Select} annotation may be applied to a {@code @Find}
+ * method that returns an entity attribute type.</li>
+ * <li>Multiple {@code @Select} annotations may be applied to a {@code @Find}
+ * method which returns a record type, one annotation for each component of
+ * the record type. The number and order of {@code @Select} annotations must
+ * match the number and order of record components.</li>
+ * <li>At most one {@code @Select} annotation may be applied to a component
+ * of a record type. When applied to a record component, the {@code Select}
+ * annotation affects every {@link Find @Find} method which returns the
+ * record type and which does not have a {@code @Select} annotation, and
+ * affects every {@link Query @Query} method which returns the record type
```

Comment:
Sorry I wasn't very clear about the grammar correction on this one. The comma can be used before "and" when the "and" joins two clauses that could be sentences of their own.  I had proposed adding "it" here so there would be a subject in the second clause.  Another option would be removing the comma, but that seemed more confusing because the sentence is already long and has another occurrences of "and" that is used differently.  In any case, this is so trivial that I'm adding the approval regardless of whether you decide to change it.

```suggestion
 * record type and which does not have a {@code @Select} annotation, and it
 * affects every {@link Query @Query} method which returns the record type
```

Reply:
> The comma can be used before "and" when the "and" joins two clauses that could be sentences of their own. 

I've been going along with this for a while now, but don't think it's actually correct. Do you have a source for it? ChatGPT thinks I'm allowed to use ", and" in this circumstance:

<img width="486" alt="Screenshot 2025-03-26 at 8 42 03 PM" src="https://github.com/user-attachments/assets/bd5114b4-af3b-4bb5-941b-27620eb2a33d" />
<img width="543" alt="Screenshot 2025-03-26 at 8 42 32 PM" src="https://github.com/user-attachments/assets/580432bf-d04e-4f07-8bed-d5a7f50bde14" />

That is to say, this is a "complex list of clauses", not just "two phrases", because each of the items has a noun and a verb, IIUC.

Reply:
I'm not expert on grammar.  When I saw it, it looked wrong, so I googled "when to use comma before and".
Google's AI told me: `Use a comma before "and" when it connects two independent clauses. An independent clause can stand alone as a sentence because it has a subject and a verb.`

Not entirely trusting AI to get things right, I also clicked on the search result,
https://www.grammarly.com/blog/punctuation-capitalization/comma-before-and/
which says,
"Use a comma before _and_ when connecting two independent clauses."
and
"Don’t use a comma before _and_ when connecting an independent clause to a dependent clause ..."

After that, I didn't look any further because it didn't seem worth it.

Reply:
I mean, I don't know English grammar either because it's not taught in Australian schools. But usually my intuition on these things is correct.

With respect to trusting language models on this sort of stuff, well, actually I trust them more than any other source. They're *really* good at this shit. I don't use them to actually _write_ stuff, because from what I've seen they tend to produce sorta bland and boring writing, but for analyzing text they're magical.

DeepSeek agrees with ChatGPT, by the way, and has an even better explanation for why this sentence was correct as originally written.

<img width="830" alt="Screenshot 2025-03-26 at 9 03 13 PM" src="https://github.com/user-attachments/assets/3077bd38-4962-40eb-98e2-71c522a3fcf5" />

Notice that DeepSeek has parsed and understood the sentence perfectly, even though this is a sufficiently grammatically complicated sentence that two _native speakers are arguing over how to punctate it_!

Reply:
It's almost like DeepSeek is claiming it's okay because the second clause becomes an independent clause due to its length and complexity rather than it actually being valid as a sentence on its own. Grammar is complex and enough time has been spent on this, so I'd say just merge it.

I'm curious what the right answer is for when this comes up again in something I'm reviewing, so maybe I'll ask our technical writers who are experts on grammar.  They review all the other documentation we write for users, but for some reason, never the specifications.

Reply:
I don't think it's an independent clause; merely a "complex clause" that occurs in a list of complex clauses. 

Remember that English is not a CFG!

Reply:
FYI - The answer I got back from our tech writer is that neither of our approaches were any good here.  The sentence is too long and should have been split in two.  If it had been a shorter sentence then the answer would have been to remove the comma before "and" because the sentence has just one subject.

Reply:
OK, fine, I put back in the colon which is actually how I had originally written this sentence.